discovery_child_development/pipeline/embed_openalex_docs.py

```diff
@@ -0,0 +1,103 @@
+import boto3
+from io import BytesIO
+import os
+from dotenv import load_dotenv
+from time import time
+import pandas as pd
+import numpy as np
+import pickle
+import sentence_transformers
+from sentence_transformers import SentenceTransformer
+
+model = SentenceTransformer("all-MiniLM-L6-v2")
+
+s3_client = boto3.client("s3")
+
+load_dotenv()
+
+S3_BUCKET = os.environ["S3_BUCKET"]
+
+CONCEPT_IDS = [
+    "C109260823",  # child development
+    "C2993937534",  # childhood development
+    "C2777082460",  # early childhood
+    "C2911196330",  # child rearing
+    "C2993037610",  # child care
+    "C2779415726",  # child protection
+    "C2781192327",  # child behavior checklist
+    "C15471489",  # child psychotherapy
+    "C178229462",  # early childhood education
+    # "C138496976",  # developmental psychology (level 1).
+]
+
+CONCEPT_IDS = "|".join(CONCEPT_IDS)
+
+
+def read_concepts_metadata(concepts_data_path: str, s3_client, s3_bucket: str):
+    """Read in the concepts metadata for all abstracts in the current OpenAlex dataset.
+
+    Args:
+        s3_client (_type_): boto3 client
+        s3_bucket (str): name of the bucket
+        concepts_data_path (str): Complete filepath within the bucket e.g. "path/to/your/data.csv"
+
+    Returns:
+        pd.DataFrame: A dataframe with the concepts metadata
+    """
+    response = s3_client.get_object(Bucket=s3_bucket, Key=concepts_data_path)
+    csv_data = response["Body"].read()
+    openalex_text_df = pd.read_csv(BytesIO(csv_data), index_col=0)
+
+    return openalex_text_df
+
+
+def upload_to_s3(
+    s3_client,
+    s3_bucket,
+    sentence_vectors_384,
+    bucket_path="inputs/data/openAlex/vectors/",
+):
+    array_data = BytesIO()
+    pickle.dump(sentence_vectors_384, array_data)
+    array_data.seek(0)
+    s3_client.upload_fileobj(
+        array_data, s3_bucket, f"{bucket_path}sentence_vectors_384.pkl"
+    )
+
+    # download in order to check that uploading does not affect the vectors in any way
+    my_array_data2 = BytesIO()
+    s3_client.download_fileobj(
+        s3_bucket, f"{bucket_path}sentence_vectors_384.pkl", my_array_data2
+    )
+    my_array_data2.seek(0)
+    my_array2 = pickle.load(my_array_data2)
+
+    # check that everything is correct
+    if not np.allclose(sentence_vectors_384, my_array2):
+        print("Warning: Arrays are not equal")
+
+
+if __name__ == "__main__":
+    openalex_text_df = read_concepts_metadata(
+        concepts_data_path=f"inputs/data/openAlex/openalex_abstracts_{CONCEPT_IDS}_year-2023.csv",
```

Comment:
I would suggest defining the hardcoded paths up top of the script as a global variable, eg:

```python
CONCEPTS_DATA_PATH = "inputs/data/openAlex/openalex_abstracts_{}_year-2023.csv"
```

Reply:
Hey, I've realised one really confusing thing in this script: I don't know why I called the function `read_concepts_metadata` - it reads in titles + abstracts, not concepts! Maybe you know this already. I'm going to amend the function name.

Reply:
Not that it's a very specialised function anyway. There's probably a better version in ds_utils (might check that as part of a separate issue)